momento-sdk/src/main/java/momento/sdk/batchutils/MomentoBatchUtils.java

```diff
@@ -0,0 +1,229 @@
+     * theory these requests should never take any disk or network time as it's simply submitting an
+     * async request, so the default 10 second timeout is a more than conservative amount for the
+     * same.
+     *
+     * @param requestTimeoutSeconds The maximum timeout for each individual request to Momento.
+     * @return The builder instance for chaining.
+     */
+    public MomentoBatchUtilsBuilder withRequestTimeoutSeconds(int requestTimeoutSeconds) {
+      this.requestTimeoutSeconds = requestTimeoutSeconds;
+      return this;
+    }
+
+    /**
+     * Builds and returns a MomentoBatchUtils instance.
+     *
+     * @return A new instance of MomentoBatchUtils.
+     */
+    public MomentoBatchUtils build() {
+      return new MomentoBatchUtils(cacheClient, maxConcurrentRequests, requestTimeoutSeconds);
+    }
+  }
+
+  public static MomentoBatchUtilsBuilder builder(final CacheClient cacheClient) {
+    return new MomentoBatchUtilsBuilder(cacheClient);
+  }
+
+  /**
+   * Performs a batch get operation for String keys.
+   *
+   * @param cacheName The name of the cache.
+   * @param request The batch get request with String keys.
+   * @return BatchGetResponse The batch get response
+   */
+  public CompletableFuture<BatchGetResponse> batchGet(
+      final String cacheName, final BatchGetRequest.StringKeyBatchGetRequest request) {
+
+    final List<CompletableFuture<BatchGetResponse.StringKeyBatchGetSummary.GetSummary>>
+        futureSummaries = new ArrayList<>();
+
+    for (final String key : request.getKeys()) {
+      final CompletableFuture<BatchGetResponse.StringKeyBatchGetSummary.GetSummary> futureSummary =
+          CompletableFuture.supplyAsync(
+              () -> {
+                try {
+                  final GetResponse getResponse =
+                      cacheClient
+                          .get(cacheName, key)
+                          .get(this.requestTimeoutSeconds, TimeUnit.SECONDS);
+                  return new BatchGetResponse.StringKeyBatchGetSummary.GetSummary(key, getResponse);
+                } catch (Exception e) {
+                  return new BatchGetResponse.StringKeyBatchGetSummary.GetSummary(
+                      key,
+                      new GetResponse.Error(CacheServiceExceptionMapper.convert(e.getCause())));
+                }
+              },
+              executorService);
+
+      futureSummaries.add(futureSummary);
+    }
+
+    // chain all the futures to generate a new future returned back to the caller
+    return CompletableFuture.allOf(futureSummaries.toArray(new CompletableFuture[0]))
+        .thenApply(
+            v -> {
+              List<BatchGetResponse.StringKeyBatchGetSummary.GetSummary> summaries =
+                  futureSummaries.stream()
+                      .map(CompletableFuture::join)
+                      .collect(Collectors.toList());
+              return new BatchGetResponse.StringKeyBatchGetSummary(summaries);
+            });
```

Comment:
Last request: could you order the keys in the summary identically to the order they are passed in?

Reply:
Same goes for the bytes summary.

Reply:
Sure, a linked hash map should do the trick